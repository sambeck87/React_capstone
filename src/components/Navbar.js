import React from 'react';
import { NavLink } from 'react-router-dom';
import { IoIosArrowBack, IoIosSettings } from 'react-icons/io';
import { BsFillMicFill } from 'react-icons/bs';
import '../styles/commons.css';
import '../styles/navbar.css';

const Navbar = () => {
  const currentYear = new Date().getFullYear();
  return (
    <div id="Navbar" className="flex row first_color lato">
      <div className="flex row">
        <NavLink
          className={({ isActive }) => (isActive ? 'activeLink' : 'homeLink')}
          to="/"
          type="button"
        >
          <IoIosArrowBack style={{ color: '#fff', fontSize: '20px', marginTop: '2px' }} />
        </NavLink>
        <p id="year">{currentYear}</p>
      </div>
      <p id="rank">Top 100 rank</p>
      <div>
        <button className="buttonNav" type="button">
          <BsFillMicFill style={{ color: '#fff', fontSize: '20px', marginTop: '2px' }} />
        </button>
        <button className="buttonNav" type="button">
          <IoIosSettings style={{ color: '#fff', fontSize: '20px', marginTop: '2px' }} />
        </button>
      </div>
    </div>
  );
};

export default Navbar;
