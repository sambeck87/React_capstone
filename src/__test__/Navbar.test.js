import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';
import '@testing-library/jest-dom';
import store from '../redux/configureStore';

describe('NavBar', () => {
  const tree = test('renders NavBar component', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
