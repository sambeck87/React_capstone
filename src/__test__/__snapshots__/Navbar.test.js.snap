// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders NavBar component 1`] = `undefined`;
